package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -930,7 +930,8 @@ def _get_timestep():
             timestep = _get_timestep()
             self.timesteps.append(timestep)
 
-            pm.echo(ts.frame)
```

Comment:
OK So after looking into the pm code it's clear why this behaves strange. The pm code as it is written expects that `ts.frame` is increasing in 1 increments but it accepts an arbitrary value.

This can be fixed within the pm class but I think we better have a short discussion before what the expected behavior actually is.